src/EventProcessor/EventProcessorInterface.js

```diff
@@ -0,0 +1,45 @@
+goog.require('measurementlibrary.Storage.StorageInterface');
+goog.module('measurementlibrary.EventProcessor.EventProcessorInterface');
+
+/**
+ * A class implementing this interface can react to events pushed to the
+ * data layer.
+ * Event processors may perform actions such as sending a network request
+ * based on the event data, interacting with the storage interface to commit
+ * and retrieve values from long-term storage, or reading values from
+ * the abstract model.
+ *
+ * @interface
+ */
+class EventProcessor {
+  /**
+   * Processes an event passed into the data layer.
+   *
+   * @param {!StorageInterface} storageInterface An interface to an object to
+   *    load or save persistent data with.
+   * @param {{get:function(string):*, set:function(string, *)}} modelInterface
+   *    An interface to load or save short term page data from the data layer.
+   * @param {string} eventName The name of the event passed to the data layer.
+   * @param {...*} eventArgs The events passed to the data layer
```

Comment:
What would an example invocation with event args look like?  I was under the impression that the eventArgs would be an optional object, rather than a varargs.

Reply:
I have it has a varargs, so implementations of the event interface could use an optional object if they want. But in theory you could invoke 
`measure('event', 'pageview', '6:00pm', 'john')`
and it would call
`processEvent(myStorage, myModel, 'pageview', '6:00pm', 'john')`

Reply:
Hmm I just read the project plan and it specified that it should be an options object explicitly. So nevermind my last comment, I edited the jsdoc. Now a call would be measure('event', 'pageview', {time:'6:00pm', name:'john'}) and then processEvent(myStorage, myModel, 'pageview', {time:'6:00pm', name:'john'}) .